edkrepo_manifest_parser/edk_manifest.py

```diff
@@ -835,8 +836,11 @@ def get_patchset_operations(self, name, remote):
         '''
         patch_set_operations = []
         if (name, remote) in self._patch_sets:
-            self.get_parent_patchset_operations(name, remote, patch_set_operations)
-            patch_set_operations.append(self._patch_set_operations[(name, remote)])
+            try:
+                self.get_parent_patchset_operations(name, remote, patch_set_operations)
+                patch_set_operations.append(self._patch_set_operations[(name, remote)])
+            except RecursionError:
+                raise ValueError(PATCHSET_PARENT_CIRCULAR_DEPENDENCY)

```

Comment:
With a PatchSet that contains circular dependencies with daisy-chained PatchSet, I can hit this exception during edkrepo checkout.

If I manually continue when the RecursionError occurs at this point, for test purposes, the other circular dependency check catches the circular dependency error:

Error: The PatchSet PatchSet(remote='Edk2Repo', name='edk2_test_c', parent_sha='edk2_test_b', fetch_branch='master') has a circular dependency with another PatchSet